src/utils/espresso/sum_check/mod.rs

```diff
@@ -9,60 +9,56 @@
 
 //! This module implements the sum check protocol.
 
-use crate::utils::virtual_polynomial::{VPAuxInfo, VirtualPolynomial};
+use crate::{
+    transcript::Transcript,
+    utils::virtual_polynomial::{VPAuxInfo, VirtualPolynomial},
+};
+use ark_ec::{CurveGroup, Group};
 use ark_ff::PrimeField;
 use ark_poly::DenseMultilinearExtension;
 use ark_std::{end_timer, start_timer};
-use std::{fmt::Debug, sync::Arc};
+use std::{fmt::Debug, marker::PhantomData, sync::Arc};
 
-use espresso_subroutines::poly_iop::{prelude::PolyIOPErrors, PolyIOP};
-use espresso_transcript::IOPTranscript;
-use structs::{IOPProof, IOPProverState, IOPVerifierState};
+use crate::utils::sum_check::structs::IOPProverMessage;
+use crate::utils::sum_check::structs::IOPVerifierState;
+use espresso_subroutines::poly_iop::prelude::PolyIOPErrors;
+use structs::{IOPProof, IOPProverState};
 
 mod prover;
 pub mod structs;
 pub mod verifier;
 
-/// Trait for doing sum check protocols.
-pub trait SumCheck<F: PrimeField> {
+/// A generic sum-check trait over a curve group
+pub trait SumCheck<C: CurveGroup> {
```

Comment:
Do we need to define it over a `CurveGroup`? I think we can make it even more generic by defining it over a Field or PrimeField and then, passing the associated curve field type. 

WDYT @arnaucube @dmpierre ?

Reply:
Yes, I think that makes more sense indeed, since it lets the user choose over which field the polynomial begin sumchecked is defined. I will also let @arnaucube chime in.

Reply:
Right, maybe one path forward is to remove the `absorb_point` from `Transcript` (and consequently from `PoseidonTranscript` too), and add it as an extension trait only for those schemes that will need to absorb points (eg. `TranscriptWithPoints`), but keeping the 'standard' transcript (and thus also the sumcheck implementation) generic over a field and not over a curvegroup.
But, the problem that I'm seeing is that `Pedersen` commitments need to absorb points to the transcript, and pedersen commitments are used in all the schemes that we have planned (at least in the initial version of them), so even that SumCheck would not need that `TranscriptWithPoints`, the schemes using the SumCheck would still require `TranscriptWithPoints`, so the transcript without absorbing points would be kind of a 'special case' used only by the SumCheck, where the rest of schemes would use `Transcript + TranscriptWithPoints`, which looks a bit of overcomplexity for the rest of the code winning only simplicity in SumCheck.

Reply:
Ohh dammit! I forgot about `absorb_point`!!!!  Forget what I said then. Let's keep it simple as it is now :pray: